core/src/style/polylineStyle.cpp

```diff
@@ -67,26 +66,160 @@ void PolylineStyle::constructShaderProgram() {
+    m_zoom = _tile.getID().z;
+    m_tileSize = _tile.getProjection()->TileSize();
+    m_mesh = std::make_unique<Mesh>(m_style.vertexLayout(), m_style.drawMode());
+    m_meshData[0].clear();
+    m_meshData[1].clear();
 }
 
-void PolylineStyle::buildPolygon(const Polygon& _poly, const DrawRule& _rule, const Properties& _props,
-                                 VboMesh& _mesh, Tile& _tile) const {
+std::unique_ptr<VboMesh> PolylineStyleBuilder::build() {
+    auto mesh = std::make_unique<Mesh>(m_style.vertexLayout(), m_style.drawMode());
+    mesh->compile(m_meshData);
+    m_meshData[0].clear();
+    m_meshData[1].clear();
+    return std::move(mesh);
+}
+
+auto PolylineStyleBuilder::parseRule(const DrawRule& _rule, const Properties& _props) -> Parameters {
+    Parameters p;
 
-    auto params = parseRule(_rule, _props, _tile);
-    for (const auto& line : _poly) {
-        buildMesh(line, params, _mesh);
+    uint32_t cap = 0, join = 0;
+
+    struct {
+        uint32_t order = 0;
+        uint32_t color = 0xff00ffff;
+        float width = 0.f;
+        float slope = 0.f;
+    } fill, stroke;
+
+    auto& width = _rule.findParameter(StyleParamKey::width);
+    if (!evalWidth(width, fill.width, fill.slope)) {
+        fill.width = 0;
+        return p;
     }
-}
+    fill.slope -= fill.width;
+    _rule.get(StyleParamKey::color, p.fill.color);
+    _rule.get(StyleParamKey::cap, cap);
+    _rule.get(StyleParamKey::join, join);
+    _rule.get(StyleParamKey::order, fill.order);
+    _rule.get(StyleParamKey::tile_edges, p.keepTileEdges);
+
+    p.fill.cap = static_cast<CapTypes>(cap);
+    p.fill.join = static_cast<JoinTypes>(join);
+
+    glm::vec2 extrude = glm::vec2(0);
+    _rule.get(StyleParamKey::extrude, extrude);
+
+    float height = getUpperExtrudeMeters(extrude, _props);
+    height *= m_tileUnitsPerMeter;
+
+    stroke.order = fill.order;
+    p.stroke.cap = p.fill.cap;
+    p.stroke.join = p.fill.join;
+
+    auto& strokeWidth = _rule.findParameter(StyleParamKey::outline_width);
+    if (strokeWidth |
+        _rule.get(StyleParamKey::outline_color, p.stroke.color) |
+        _rule.get(StyleParamKey::outline_order, stroke.order) |
+        _rule.get(StyleParamKey::outline_cap, cap) |
+        _rule.get(StyleParamKey::outline_join, join)) {
+
+        p.stroke.cap = static_cast<CapTypes>(cap);
+        p.stroke.join = static_cast<JoinTypes>(join);
+
+        if (!evalWidth(strokeWidth, stroke.width, stroke.slope)) {
+            return p;
+        }
+
+        // NB: Multiply by 2 for the stroke to get the expected stroke pixel width.
+        stroke.width *= 2.0f;
+        stroke.slope *= 2.0f;
+        stroke.slope -= stroke.width;
+
+        stroke.width += fill.width;
+        stroke.slope += fill.slope;
 
-void PolylineStyle::buildLine(const Line& _line, const DrawRule& _rule, const Properties& _props,
-                              VboMesh& _mesh, Tile& _tile) const {
+        stroke.order = std::min(stroke.order, fill.order);
```

Comment:
stroke is uint in this place

Just noted again: Is everybody OK with using stroke as name instead of outline? I thought to have changed it back before committing..